### PR TITLE
DAOS-2781 build: upgrade SPDK to v19.04.1

### PR DESCRIPTION
**Pull request by NiuYawei**
Upgrade SPDK to v19.04.1

Signed-off-by: Niu Yawei <yawei.niu@intel.com>

### CODE REVIEW

File: sl_patches/spdk_export_set_thread.patch [hunk 1]
```diff
@@ -0,0 +1,33 @@
+diff --git a/include/spdk/thread.h b/include/spdk/thread.h
+index 64b04be..50b473a 100644
+--- a/include/spdk/thread.h
++++ b/include/spdk/thread.h
+@@ -206,6 +206,13 @@ void spdk_thread_lib_fini(void);
+ struct spdk_thread *spdk_thread_create(const char *name, struct spdk_cpuset *cpumask);
```
Comment (jolivier23) [inline sl_patches/spdk_export_set_thread.patch]: So are we using an API that isn't exported?   Have we reported to SPDK team?

Comment (NiuYawei) [reply sl_patches/spdk_export_set_thread.patch]: This API is already exported in latest SPDK. Yes, your proposal of changing DAOS looks good to me. I'll update it soon.

File: prereq_tools/base.py [hunk 2]
```diff
@@ -816,10 +816,8 @@ def _setup_intelc(self):
         version = env.get("INTEL_C_COMPILER_VERSION")
         self.__env.Replace(INTEL_C_COMPILER_VERSION=version)
         self.__env.Replace(LINK=env.get("LINK"))
-        # Link intel symbols statically with no external visibility and
         # disable the warning about Cilk since we don't use it
-        self.__env.AppendUnique(LINKFLAGS=["-Wl,--exclude-libs,ALL",
-                                           "-static-intel",
+        self.__env.AppendUnique(LINKFLAGS=["-static-intel",
```
Comment (jolivier23) [inline prereq_tools/base.py]: I think this will cause some checks to fail

**Comment by jolivier23** [comment]
> Shall we wait on landing this until the daos patch that uses this has completed and is ready to land also, just to avoid a need to back this out again to move scons_local before the doas patch is ready to land?

Yeah, that seems sensible.   Where are we at with the DAOS patch?

**Pull request merged**